libs/SalesforceSDK/src/com/salesforce/androidsdk/accounts/UserAccountBuilder.kt

```diff
@@ -0,0 +1,529 @@
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission of salesforce.com, inc.
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.salesforce.androidsdk.accounts
+
+import com.salesforce.androidsdk.auth.OAuth2.IdServiceResponse
+import com.salesforce.androidsdk.auth.OAuth2.TokenEndpointResponse
+
+/**
+ * This class helps build a UserAccount object.
+ */
+class UserAccountBuilder private constructor() {
+    private var authToken: String? = null
+    private var refreshToken: String? = null
+    private var loginServer: String? = null
+    private var idUrl: String? = null
+    private var instanceServer: String? = null
+    private var orgId: String? = null
+    private var userId: String? = null
+    private var username: String? = null
+    private var accountName: String? = null
+    private var communityId: String? = null
+    private var communityUrl: String? = null
+    private var firstName: String? = null
+    private var lastName: String? = null
+    private var displayName: String? = null
+    private var email: String? = null
+    private var photoUrl: String? = null
+    private var thumbnailUrl: String? = null
+    private var lightningDomain: String? = null
+    private var lightningSid: String? = null
+    private var vfDomain: String? = null
+    private var vfSid: String? = null
+    private var contentDomain: String? = null
+    private var contentSid: String? = null
+    private var csrfToken: String? = null
+    private var nativeLogin = false
+    private var language: String? = null
+    private var locale: String? = null
+    private var cookieClientSrc: String? = null
+    private var cookieSidClient: String? = null
+    private var sidCookieName: String? = null
+    private var clientId: String? = null
+    private var additionalOauthValues: Map<String, String>? = null
+    private var allowUnset = true
+
+    /**
+     * Set fields from token end point response
+     * @param tr token endpoint response
+     * @return Instance of this class
+     */
+    fun populateFromTokenEndpointResponse(tr: TokenEndpointResponse?): UserAccountBuilder {
```

Comment:
Key method to get values from token end point response into a user account object.